File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagram_dict = defaultdict(list)

        for s in strs:
            key = tuple(sorted(s))
            anagram_dict[key].append(s)

        return list(anagram_dict.values())

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> anagram_map;

        for (const auto& s : strs) {
            string key = s;
            sort(key.begin(), key.end()); 

            anagram_map[key].push_back(s);
        }

        vector<vector<string>> group_anagrams;
        for (const auto& pair : anagram_map) {
            group_anagrams.push_back(pair.second);
        }

        return group_anagrams;
    }
};
